size_of_longest_substring.cpp

```diff
@@ -0,0 +1,138 @@
+        }
+        else // if character is repeated
+        {
+            char tempc = s.at(count);
+            int tempi = s.find(tempc);
+            s = s.substr(tempi+1);
+            count = 0;
+            if(max < (int)current_substring.size() )
+            {
+                max = (int)current_substring.size();
+            }
+            current_substring = "";
+        }
+        
+    }
+    
+    if( max < (int)current_substring.size())
+    {
+        max = (int)current_substring.size();
+    }
+    
+    return max;
+} 
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    /* Test 1: 
+    s = "abcabcbb" 
+    returns 3 */
+    std::cout << "Test 1:\n";
+    std::string s = "abcabcbb";
+    assert(find_longest_substring(s) == 3); // this ensures that the algorithm works as expected
+    std::cout << "String \"abcabcbb\" will return 3.\n";
+    
+    /* Test 2: 
+    s = ""
+    returns 0 */
+    std::cout << "Test 2:\n";
+    s = "";
+    assert(find_longest_substring(s) == 0); // this ensures that the algorithm works as expected
+    std::cout << "String \"\" will return 0.\n";
+    
+    /* Test 3: 
+    s= " "
+    returns 1 */
+    std::cout << "Test 3:\n";
+    s = " ";
+    assert(find_longest_substring(s) == 1); // this ensures that the algorithm works as expected
+    std::cout << "String \" \" will return 1.\n";
+    
+    /* Test 4: 
+    s= "ckcr"
+    returns 3 */
+    std::cout << "Test 4:\n";
+    s = "ckcr";
+    assert(find_longest_substring(s) == 3); // this ensures that the algorithm works as expected
+    std::cout << "String \"ckcr\" will return 3.\n";
+    
+    /* Test 5: 
+    s = "nvskbea"
+    returns 7 */
+    std::cout << "Test 5:\n";
+    s = "nvskbea";
+    assert(find_longest_substring(s) == 7); // this ensures that the algorithm works as expected
+    std::cout << "String \"nvskbea\" will return 7.\n";
+    
+    /* Test 6: 
+    s = "wwwwww"
+    returns 1 */
+    std::cout << "Test 6:\n";
+    s = "wwwwww";
+    assert(find_longest_substring(s) == 1); // this ensures that the algorithm works as expected
+    std::cout << "String \"wwwwww\" will return 1.\n";
+    
+    /* Test 7: 
+    s = "rrewasr"
+    returns 5 */
+    std::cout << "Test 7:\n";
+    s = "rrewasr";
+    assert(find_longest_substring(s) == 5); // this ensures that the algorithm works as expected
+    std::cout << "String \"rrewasr\" will return 5.\n";
+    
+    /* Test 8: 
+    s= "pwwkew"
+    returns 3 */
+    std::cout << "Test 8:\n";
+    s = "pwwkew";
+    assert(find_longest_substring(s) == 3); // this ensures that the algorithm works as expected
+    std::cout << "String \"pwwkew\" will return 3.\n";
+}
+
+/**
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns 0 on exit
+ */
+int main(int argc, char *argv[]) {
+    test();  // run self-test implementations
+    return 0;
+}
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```